src/core_plugins/elasticsearch/index.js

```diff
@@ -15,8 +16,6 @@ const DEFAULT_REQUEST_HEADERS = [ 'authorization' ];
 
 module.exports = function ({ Plugin }) {
   return new Plugin({
-    require: ['kibana'],
```

Comment:
That's tricky... thinking about these dependencies led me to another question as well: Is there a possible race condition between mapping registration and .kibana index creation? The .kibana index is created by the health check, which gets kicked off in the elasticsearch plugin's init function. If the Kibana plugin registers its mappings in its own init function, and it requires the elasticsearch plugin meaning the ES plugin's init function will run first, isn't there a chance the .kibana index will get created before the kibana plugin has a chance to register its mappings? If I'm not misunderstanding something, it seems this only works now because the healthCheck's start method has a short timeout.

To avoid both of these issues, what if we made the mappings a part of the actual plugin definition? All of the plugins get registered before any get initialized, so that would guarantee all mappings have been registered by the time the ES plugin is initializing, and the kibana plugin wouldn't need to `require` the elasticsearch plugin.